tezedge_state/src/proposer/proposal_loader.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use serde_json::de::IoRead;
+use serde_json::StreamDeserializer;
+use std::fs::File;
+use std::io::BufReader;
+
+use crate::proposals::*;
+
+pub struct ProposalLoader {
+    reader: StreamDeserializer<'static, IoRead<BufReader<File>>, RecordedProposal>,
+}
+
+impl ProposalLoader {
+    pub fn new() -> Self {
+        let file = File::open("/tmp/tezedge/recorded_proposals.log")
+            .expect("couldn't open file for loading proposals");
+        let reader = serde_json::Deserializer::from_reader(BufReader::new(file)).into_iter();
+        Self { reader }
+    }
+
+    pub fn next(&mut self) -> Option<Result<RecordedProposal, serde_json::Error>> {
+        self.reader.next()
+    }
+}
+
+impl Clone for ProposalLoader {
+    fn clone(&self) -> Self {
+        // TODO
```

Comment:
@binier 
do we still need that?

Reply:
Yes. It's kinda a dirty trick to make it work for testing, because we need cloning in testing and there we don't need record/replay functionality, so that clone won't ever be called.

In real run, whenever we use `mio` types from that generic argument is not clonable, so we can't ever clone `TezedgeProposer` and hit that code.

Basically once I make recorder/replayer generic types on proposer, those can be removed.

Reply:
Removed.